### PR TITLE
feat: Add AtLeastOne type

### PR DESCRIPTION
**Pull request by Drodevbar**
## Changes

I find this util type useful when I have a function that requires me to pass at least one param. Thanks to it, I don't have to manually verify whether one of them was passed, but TS can do the work.

## Checklist

- [x] Apply one of following labels; `major`, `minor`, `patch` or `skip-release`
- [x] I've updated the documentation, or no changes were necessary
- [x] I've updated the tests, or no changes were necessary

### CODE REVIEW

File: src/common/atLeastOne.spec.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    interface TestType {
+      a: string
+      b: number
+    }
+
+    const testCase1: AtLeastOne<TestType> = { a: 'a' }
+    const testCase2: AtLeastOne<TestType> = { b: 1 }
+
+    assertType<{ a: string; b?: number }>(testCase1)
+    assertType<{ a?: string; b: number }>(testCase2)
```
Comment (kibertoad) [inline src/common/atLeastOne.spec.ts:15]: can you also add negative case?

File: src/common/atLeastOne.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+// can be used when you want to make at least one key of an object required
+export type AtLeastOne<T, U = { [K in keyof T]: Pick<T, K> }> = Partial<T> & U[keyof U]
```
[the remark this answers is not in the document]
Comment (Drodevbar) [reply src/common/atLeastOne.ts:1]: I've added them to `docs/custom-types.md` as I see we have some documents already there. I've included the existing types that haven't been documented so far.

File: src/common/atLeastOne.spec.ts (hunk 1 above)
Comment (Drodevbar) [reply src/common/atLeastOne.spec.ts:15]: Added negative test ✅ 

I've also enabled `typecheck` as by default vitest has it disabled - please see `vitest.config.mts`

**Review by kibertoad — APPROVED** [review]
thank you, this is great!

**Pull request merged**